pageserver/src/tenant/timeline.rs

```diff
@@ -4294,7 +4324,15 @@ impl Timeline {
             }
 
             if let Some(remote_client) = &self.remote_client {
-                remote_client.schedule_layer_file_deletion(&layer_names_to_delete)?;
+                // Remote metadata upload was scheduled in `update_metadata_file`: wait
+                // for completion before scheduling any deletions.
+                remote_client.wait_completion().await?;
```

Comment:
> That is correct, but, 2 wait_completion calls instead of 0 wait_completion calls before.

1 before, right?  The wait_completion in the "waiting for upload ops to complete" part above was already there.

The ideal procedure here depends how leak-proof we want to be.  It is sufficient for correctness to do:
1. Update metadata to advance cutoff and remove layers below cutoff
2. wait_completion to flush the write in 1
3. push to deletion queue

Or, to avoid a window where we might forget about some objects on a crash, we could do:
1. Update metadata to advance cutoff, and require that any readers do not read below that cutoff so that these objects become safe to delete (not sure if we do that currently?)
2. Flush 1
3. Push objects to deletion queue & flush
4. Update metadata to remove objects
5. Flush 4

Getting rid of the initial "waiting for upload ops to complete" wait_completion feels like the main sticking point here, and that would require some smarter cancellation/wait logic in remote_timeline_client